### PR TITLE
[DOPS-1242] Git Actions CI added

### PR DESCRIPTION
**Pull request by Cre-eD**
# Task

[DOPS-1242]: Set up Github Actions CI.

## Changes

1. Github Actions Ci configured and tested.
2. Sonarqube check configured.
3. Codecov check configured.

## Tests

[Github Actions **CI**]

**Sonarqube** report:

image

**Codecov** report:

image

## Author

Signed-off-by: Dmitriy Creed <creed@soramitsu.co.jp>

[Github Actions **CI**]: https://github.com/soramitsu/iroha-java/pull/1/checks
[DOPS-1242]: https://soramitsu.atlassian.net/browse/DOPS-1242

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+          ./gradlew --daemon
+          ./gradlew build --parallel --build-cache
+          ./gradlew jacocoTestReport
+      - name: Cleanup Gradle Cache
+        run: |
+          rm -f ~/.gradle/caches/modules-2/modules-2.lock
+          rm -f ~/.gradle/caches/modules-2/gc.properties
+      - name: Codecov
+        env:
+          CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
```
Comment (BulatSaif) [inline .github/workflows/ci.yml:42]: use codecov module https://github.com/marketplace/codecov, it doesn't need CODECOV_TOKEN
example: https://github.com/libp2p/cpp-libp2p/pull/112/files#diff-a2115d277b5ca5a2f09a999e53440839cf332b94da177f3d1766334555b0f7c6R40

**Review by ravenolf — APPROVED** [review]

**Review by Mingela — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Cre-eD) [reply .github/workflows/ci.yml]: CI updated

**Review by BulatSaif — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
+  CACHE_VERSION: v01
+  CACHE_PATHS: |
+    ~/.gradle/caches
+    ~/.gradle/wrapper
+jobs:
+  Linux:
+    strategy:
+      fail-fast: false
+    runs-on: ubuntu-16.04
```
Comment (kogeler) [inline .github/workflows/ci.yml:18]: Why do you need such old ubuntu? 16.04 has end of life on April 2021. You will have a problem with JDK security updates!

**Review by kogeler — APPROVED** [review]
OK. But see the comment.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (Cre-eD) [inline .github/workflows/ci.yml:18]:
```suggestion
    runs-on: ubuntu-20.04
```

**Pull request merged**